spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/http/codec/JacksonCodecConfiguration.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.http.codec;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.codec.CodecConfigurer;
+import org.springframework.http.codec.json.Jackson2JsonDecoder;
+import org.springframework.http.codec.json.Jackson2JsonEncoder;
+import org.springframework.util.MimeType;
+
+/**
+ * Configuration for HTTP codecs that use Jackson.
+ * @author Brian Clozel
+ */
+@Configuration
+@ConditionalOnClass(ObjectMapper.class)
+class JacksonCodecConfiguration {
+
+	@Configuration
+	@ConditionalOnBean(ObjectMapper.class)
+	protected static class JacksonJsonCodecConfiguration {
```

Comment:
The extra inner class shouldn't be necessary. The conditional on bean can be moved to the bean definition.